sycl/test-e2e/Annotated_usm/annotated_usm.cpp

```diff
@@ -0,0 +1,173 @@
+             properties{alignment<0>});
+  TEST_ALIGN((malloc_host_annotated<T>), sizeof(T), N, q,
+             properties{alignment<0>});
+  if (dev.has(sycl::aspect::usm_shared_allocations)) {
+    TEST_ALIGN((malloc_shared_annotated<T>), sizeof(T), N, q,
+               properties{alignment<0>});
+  }
+
+  // Case: malloc_xxx with compile-time alignment<N> (N is not a power of 2),
+  // nullptr is returned
+  TEST_ALIGN_NULL_EXPECTED((malloc_device_annotated<T>), N, q,
+                           properties{alignment<3>});
+  TEST_ALIGN_NULL_EXPECTED(malloc_device_annotated, N, q,
+                           properties{alignment<7>});
+  TEST_ALIGN_NULL_EXPECTED((malloc_host_annotated<T>), N, q,
+                           properties{alignment<15>});
+  TEST_ALIGN_NULL_EXPECTED(malloc_host_annotated, N, q,
+                           properties{alignment<7>});
+  if (dev.has(sycl::aspect::usm_shared_allocations)) {
+    TEST_ALIGN_NULL_EXPECTED((malloc_shared_annotated<T>), N, q,
+                             properties{alignment<31>});
+    TEST_ALIGN_NULL_EXPECTED(malloc_shared_annotated, N, q,
+                             properties{alignment<63>});
+  }
+
+  // Case: malloc_xxx with compile-time alignment<N> (N is a power of 2), the
+  // alignment is the least-common-multiple of N and sizeof(T)
+  TEST_ALIGN((malloc_device_annotated<T>), std::lcm(2, sizeof(T)), N, q,
+             properties{alignment<2>});
+  TEST_ALIGN(malloc_device_annotated, 2, N, q, properties{alignment<2>});
+  TEST_ALIGN((malloc_host_annotated<T>), std::lcm(8, sizeof(T)), N, q,
+             properties{alignment<8>});
+  TEST_ALIGN(malloc_host_annotated, 8, N, q, properties{alignment<8>});
+  if (dev.has(sycl::aspect::usm_shared_allocations)) {
+    TEST_ALIGN((malloc_shared_annotated<T>), std::lcm(16, sizeof(T)), N, q,
+               properties{alignment<16>});
+    TEST_ALIGN(malloc_shared_annotated, 16, N, q, properties{alignment<16>});
+  }
+
+  // Case: aligned_alloc_xxx with no alignment property, and the alignment
+  // argument is 0 the result is sizeof(T)-aligned
+  TEST_ALIGN((aligned_alloc_device_annotated<T>), sizeof(T), 0, N, q);
+  TEST_ALIGN((aligned_alloc_host_annotated<T>), sizeof(T), 0, N, q);
+  if (dev.has(sycl::aspect::usm_shared_allocations)) {
+    TEST_ALIGN((aligned_alloc_shared_annotated<T>), sizeof(T), 0, N, q);
+  }
+
+  // Case: aligned_alloc_xxx with no alignment property, and the alignment
+  // argument is not a power of 2, the result is nullptr
+  TEST_ALIGN_NULL_EXPECTED(aligned_alloc_device_annotated, 3, N, q);
+  TEST_ALIGN_NULL_EXPECTED((aligned_alloc_host_annotated<T>), 7, N, q);
+  if (dev.has(sycl::aspect::usm_shared_allocations)) {
+    TEST_ALIGN_NULL_EXPECTED((aligned_alloc_shared_annotated<T>), 15, N, q);
+  }
+
+  // Case: aligned_alloc_xxx with compile-time alignment<N> (N is a power of 2),
+  // the alignment is the least-common-multiple of the alignment argument (a
+  // power of 2), N and sizeof(T)
```

Comment:
"the alignment is the least-common-multiple of the alignment argument (a power of 2), N and sizeof(T)"

Using 'N' in the above comment is confusing because the program contains N but it is the number of bytes. Maybe use 'k' instead of 'N' in this comment.

Reply:
comment updated